pkg/providers/nutanix/template.go

```diff
@@ -568,3 +574,118 @@ func generateNutanixFailureDomains(eksNutanixFailureDomains []v1alpha1.NutanixDa
 	}
 	return failureDomains
 }
+
+func incrementIP(ip net.IP) {
+	for i := len(ip) - 1; i >= 0; i-- {
+		ip[i]++
+		if ip[i] > 0 {
+			break
+		}
+	}
+}
+
+func compareIP(ip1, ip2 net.IP) (int, error) {
+	if len(ip1) != len(ip2) {
+		return -1, fmt.Errorf("IP addresses are not the same protocol")
+	}
+
+	for i := 0; i < len(ip1); i++ {
+		if ip1[i] < ip2[i] {
+			return -1, nil
+		}
+		if ip1[i] > ip2[i] {
+			return 1, nil
+		}
+	}
+
+	return 0, nil
+}
+
+func addCIDRToIgnoredNodeIPsList(cidr string, result []string) []string {
+	ip, ipNet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		// log error and continue
+		log.Printf("error parsing CIDR %s: %v", cidr, err)
+		return result
+	}
+
+	// Add all ip addresses in the range to the list
+	for ip := ip.Mask(ipNet.Mask); ipNet.Contains(ip); incrementIP(ip) {
+		if ip != nil {
+			result = append(result, ip.String())
+		}
+	}
+
+	return result
+}
+
+func addIPRangeToIgnoredNodeIPsList(ipRangeStr string, result []string) []string {
+	// Parse the range
+	ipRange := strings.Split(ipRangeStr, "-")
+	if len(ipRange) != 2 {
+		// log error and return
+		log.Printf("error parsing range %s: expected 2 values, got %d", ipRangeStr, len(ipRange))
+		return result
+	}
+
+	// Parse the start and end of the range
+	start := net.ParseIP(strings.TrimSpace(ipRange[0]))
+	end := net.ParseIP(strings.TrimSpace(ipRange[1]))
+	if start == nil || end == nil {
+		// log error and return
+		log.Printf("error parsing range %s: invalid IP address", ipRangeStr)
+		return result
+	}
+
+	cmp, err := compareIP(start, end)
+	if err != nil {
+		// log error and return
+		log.Printf("error comparing IP addresses %s and %s: %v", start.String(), end.String(), err)
+		return result
+	}
+
+	if cmp >= 0 {
+		// swap start and end if start is greater than end
+		start, end = end, start
+	}
```

Comment:
same as above, already validated in validator, can be removed

Reply:
Fixed